src/test/java/io/knotx/commons/http/request/DataObjectsUtilTest.java

```diff
@@ -0,0 +1,152 @@
+    public static final String[] CONTENT_TYPE_HTML = {
+            "text/html",
+            "charset=UTF-8"
+    };
+    private static final String[] CONTENT_TYPE_HTML_CLONE = CONTENT_TYPE_HTML.clone();
+    private static final String[] CONTENT_TYPE_REVERSE = {
+            "charset=UTF-8",
+            "text/html"
+    };
+
+    private static final String JSON_BODY_BASE64 =
+            "ewogICJvYmplY3QiOiB7CiAgICAiYSI6ICJiIiwKICAgICJjIjogImQiLAogICAgImUiOiAiZiIKICB9LAog" +
+            "ICJhcnJheSI6IFsKICAgIDEsCiAgICAyCiAgXSwKICAic3RyaW5nIjogIkhlbGxvIFdvcmxkIgp9";
+    private static final String OTHER_JSON_BODY_BASE64 =
+            "ewogICJvYmplY3QiOiB7CiAgICAiYSI6ICJiIiwKICAgICJjIjogImQiLAogICAgImUiOiAiZiIKICB9Cn0";
+
+    @Test
+    @DisplayName("Expect two empty elements are the same")
+    void equalsEmptyElementsAreTheSame() {
+        MultiMap multi1 = MultiMap.caseInsensitiveMultiMap();
+        MultiMap multi2 = MultiMap.caseInsensitiveMultiMap();
+        assertTrue(DataObjectsUtil.equalsMultiMap(multi1, multi2));
+
+        multi2.add("Content-Type", "text/html");
+        assertFalse(DataObjectsUtil.equalsMultiMap(multi1, multi2));
+    }
+
+    @Test
+    @DisplayName("Expect two case insensitive elements are the same")
+    void expectCaseInsensitiveElementsAreTheSame() {
+        MultiMap multi1 = MultiMap.caseInsensitiveMultiMap();
+        MultiMap multi2 = MultiMap.caseInsensitiveMultiMap();
+
+        multi1.add("content-type", "text/html");
+        multi2.add("Content-Type", "text/html");
+
+        assertTrue(DataObjectsUtil.equalsMultiMap(multi1, multi2));
+
+        multi2.add("Content-Type", "multipart/form-data");
+        assertFalse(DataObjectsUtil.equalsMultiMap(multi1, multi2));
+    }
+
+    @Test
+    @DisplayName("Expect multi value elements are the same if they keep order")
+    void expectMultiValueElementsAreTheSameIfKeepOrder() {
+        MultiMap multi1 = MultiMap.caseInsensitiveMultiMap();
+        MultiMap multi2 = MultiMap.caseInsensitiveMultiMap();
+
+        List<String> contentType1 = Arrays.asList(CONTENT_TYPE_HTML);
+        List<String> contentType2 = Arrays.asList(CONTENT_TYPE_HTML_CLONE);
+
+        multi1.add("content-type", contentType1);
+        multi2.add("Content-Type", contentType2);
+        assertTrue(DataObjectsUtil.equalsMultiMap(multi1, multi2));
+    }
+
+    @Test
+    @DisplayName("Expect multi value elements aren't the same if they don't keep order")
+    void expectMultiValueElementsNotSameIfDontKeepOrder() {
+        MultiMap multi1 = MultiMap.caseInsensitiveMultiMap();
+        MultiMap multi2 = MultiMap.caseInsensitiveMultiMap();
+
+        List<String> contentType1 = Arrays.asList(CONTENT_TYPE_HTML);
+        List<String> contentTypeReverse = Arrays.asList(CONTENT_TYPE_REVERSE);
+
+        multi1.add("content-type", contentType1);
+        multi2.add("Content-Type", contentTypeReverse);
+        assertFalse(DataObjectsUtil.equalsMultiMap(multi1, multi2));
```

Comment:
It looks, that additionally case-sensitivity is not kept. Can you please make the key equal?

Reply:
Done